[1-3] rk1

DeadBread-001 opened the pull request.

DeadBread-001 committed:
[3] added films section

Tovardas committed:
i wanna fetch

DeadBread-001 committed:
[1-2] remove/add buttons when logging

Kand1 commented:
По-моему у вас нет понимания, что такое в фронтенд проекте папка public, погуглите почитайте) Переименуйте пока в src, потом разберетесь, если надо будет добавите

DeadBread-001 committed:
from ajax to fetch

DeadBread-001 committed:
fixed server and send requests to back

Tovardas committed:
new package.json

Tovardas committed:
validation added + back port

Tovardas committed:
front port fix

DeadBread-001 committed:
fixed front port

DeadBread-001 committed:
fixe import validator

DeadBread-001 committed:
added cors

DeadBread-001 committed:
fix parsing

DeadBread-001 committed:
Revert "fix parsing"

DeadBread-001 committed:
another fixes of ports and parsing

DeadBread-001 committed:
Merge remote-tracking branch 'origin/1-3-rk1' into 1-3-rk1

DeadBread-001 committed:
right api

DeadBread-001 committed:
trying to fix

DeadBread-001 committed:
parsing fixes

DeadBread-001 committed:
another fixes

DeadBread-001 committed:
another fixes

DeadBread-001 committed:
base64 fixes

DeadBread-001 committed:
fetch fixes

DeadBread-001 committed:
logout fix

DeadBread-001 committed:
fix responses

DeadBread-001 committed:
check response

DeadBread-001 committed:
revert response

DeadBread-001 committed:
revert response

DeadBread-001 committed:
response fixes

DeadBread-001 committed:
visibility fixes

DeadBread-001 committed:
visibility fixes

DeadBread-001 committed:
visibility fixes + images

Tovardas committed:
Merge branch '1-3-rk1' of https://github.com/frontend-park-mail-ru/2024_1_Netrunners into 1-3-rk1

fake)_pull

Tovardas committed:
Merge branch '1-3-rk1' of https://github.com/frontend-park-mail-ru/2024_1_Netrunners into 1-3-rk1

Tovardas committed:
no passw_conf in signup

Tovardas committed:
working ver from serv

DeadBread-001 committed:
show film title

DeadBread-001 committed:
Merge remote-tracking branch 'origin/1-3-rk1' into 1-3-rk1

DeadBread-001 committed:
show film title

DeadBread-001 committed:
Исправил стили

DeadBread-001 committed:
color of title

DeadBread-001 committed:
another fixes

DeadBread-001 committed:
check css

DeadBread-001 committed:
fix css

DeadBread-001 committed:
fix css

DeadBread-001 committed:
fix css

DeadBread-001 committed:
fix css

DeadBread-001 committed:
fix css

DeadBread-001 committed:
debug

DeadBread-001 committed:
debug

Tovardas committed:
new style fil-cards

Tovardas committed:
Merge remote-tracking branch 'origin/1-3-rk1' into 1-3-rk1

Tovardas committed:
check_fix

Tovardas committed:
check_fix

Tovardas committed:
sign_up return added and valid_login

Tovardas committed:
sign_up return added and valid_login

Tovardas committed:
RenderMenu response.json instead of .text

Tovardas committed:
RenderMenu response.json instead of .text

Tovardas committed:
response.json into response in renderMenu

Tovardas committed:
response.json into response in renderMenu

Tovardas committed:
final_fixes?

Tovardas committed:
final_fixes?

Tovardas committed:
localstorage

Tovardas committed:
localstorage

Tovardas committed:
cookie write

Tovardas committed:
cookie write

Tovardas committed:
let token

Tovardas committed:
let token

DeadBread-001 committed:
check signup

DeadBread-001 committed:
check signup

DeadBread-001 committed:
Исправлена обработка ответа на запросы + переадресация

DeadBread-001 committed:
Исправлена обработка ответа на запросы + переадресация

Kand1 commented:
1. Все обращение к апи вынести в папку апи в отдельные функции

Kand1 commented:
2. Разнести рендер страниц из файла index.js в соответствующие компоненты

Kand1 commented:
3. Вынести все цвета и шрифты в css в осознанно названные переменные

Kand1 commented:
4. Сделать все отступы стандартизированными 2 4 8 16 32 48 64 px (паддинги марджины и тд)

Kand1 commented:
5. убарть из index.css все кроме root и body, где только переменные и стии на весь проект, остальное в папочки компонентов в css файлы

DeadBread-001 committed:
Merge branch '1-3-rk1' of https://github.com/frontend-park-mail-ru/2024_1_Netrunners into 1-3-rk1

DeadBread-001 committed:
Разделение на компоненты

Временно добавил возможность запускать сервера локально + разделил index.js на компоненты

DeadBread-001 committed:
Добавил линтер и исправил ошибки

DeadBread-001 committed:
Добавил логотип, иконки и исправил рендер меню

DeadBread-001 committed:
Убрал лишнюю функцию и исправил рендер меню

DeadBread-001 committed:
Исправил стили + убрал лишний ипорт

Tovardas committed:
Страницы с помощью шаблонов

Kand1 commented on .gitignore:
```diff
@@ -0,0 +1,3 @@
+node_modules
+.idea
+src/components/Menu/*.precompiled.js
```
Почему игнор только в папке menu? Нужно на все *.precompiled.js

Kand1 commented on src/api/auth.js:
```diff
@@ -0,0 +1,24 @@
+import {fetchRequest} from './fetch.js';
+
+const localIP = 'http://127.0.0.1:8081/';
+const globalIP = 'http://94.139.247.246:8081/';
```
У вас в каждом файле апи эти константы, вынесите их куда то в одно место, что б они не повторялись. И CONSTANT_CASE юзайте для них

Kand1 commented on src/components/Films/films.js:
```diff
@@ -0,0 +1,35 @@
+            duration: timeConvert.timeIntoText(film.duration)
+          }))
+            document.querySelector('main').innerHTML = template({filmsWithHours});
+            console.log(filmsSection);
+            console.log(filmsSection[0]);
```
Уберите все консоль логи здесь и дальше, их в main не должно быть

Kand1 commented on src/components/Films/films.js:
```diff
@@ -0,0 +1,35 @@
+      .then((response) => {
+        if (response.ok) {
+          return response.json();
+        } else {
```
else не нужен

Kand1 commented on src/components/Films/films.js:
```diff
@@ -0,0 +1,35 @@
+            document.querySelector('main').innerHTML = template({filmsWithHours});
+            console.log(filmsSection);
+            console.log(filmsSection[0]);
+          return filmsSection[0];
+        } else {
```
else не нужен

Kand1 commented on src/components/Login/login.js:
```diff
@@ -0,0 +1,45 @@
+    authApi.login(user)
+        .then((response) => {
+          if (response.ok) {
+              return response.json();
+          } else {
```
else не нужен

Kand1 commented on src/components/Logout/logout.js:
```diff
@@ -0,0 +1,24 @@
+  authApi.logout()
+      .then((response) => {
+        if (response.ok) {
+          return response.json();
+        } else {
```
else не нужен

Kand1 commented on src/components/Logout/logout.js:
```diff
@@ -0,0 +1,24 @@
+      })
+      .then((response) => {
+          updateMenuDisplay(!response.status);
+          if (response.status === 200) goToPage(menu.state.menuElements.films);
```
Давайте писать if всегда с {} Так читабельнее. Нужно будет линтер потом на это настроить. И давайте вынесем

Kand1 commented on src/components/Menu/Menu.js:
```diff
@@ -0,0 +1,40 @@
+    const template = Handlebars.templates['Menu.hbs'];
+    const items = this.items.map(([key, {href, text}], index) => {
+      let className = 'menu-item';
+      if (index === 0) {
```
Можно просто !index

Kand1 commented on src/components/Profile/profile.js:
```diff
@@ -0,0 +1,27 @@
+  authApi.check()
+      .then((response) => {
+        if (response.ok) {
+          return response.json();
+        } else {
```
else не нужен

Kand1 commented on src/components/Profile/profile.js:
```diff
@@ -0,0 +1,27 @@
+        }
+      })
+      .then((response) => {
+        if (response.status === 200) {
+        } else if (response.status === 401) {
```
Зачем здесь if с пустым телом?

Kand1 commented on src/components/Signup/signup.js:
```diff
@@ -0,0 +1,59 @@
+        .then((response) => {
+          if (response.ok) {
+            return response.json();
+          } else {
```
else не нужен

Kand1 commented on src/components/Signup/signup.js:
```diff
@@ -0,0 +1,59 @@
+        })
+        .then((response) => {
+          renderMenu();
+          if (response.status === 200) {
+            goToPage(menu.state.menuElements.films);
+          } else if (response.status === 400) {
+            throw new Error('Неверная почта или пароль при регистрации');
+          }
+        })
```
Поменять местами if-ы, и можно без else

Kand1 commented on src/index.js:
```diff
@@ -0,0 +1,90 @@
+  authApi.check()
+      .then((response) => {
+        if (response.ok) {
+          return response.json();
+        } else {
```
else не нужен

Kand1 commented on src/utils/timeConvert.js:
```diff
@@ -0,0 +1,19 @@
+};
+
+const timeIntoText = (durationInSec) => {
+    let hours = Math.floor(durationInSec / 3600);
+    let minutes = Math.floor((durationInSec % 3600) / 60);
```
тут должны быть const, раз они не меняются, на это тоже надо линтер поставить

DeadBread-001 committed:
Множество исправлений и рефакторинг

DeadBread-001 committed:
Исправил константу IP

Tovardas committed:
Добавил JSDoc и favicon + рефакторинг

DeadBread-001 committed:
Поменял IP
